frontend/public/components/instantiate-template.tsx

```diff
@@ -221,16 +256,19 @@ class TemplateForm_ extends React.Component<
     this.setState({ error: '', inProgress: true });
     this.createTemplateSecret()
       .then((secret: K8sResourceKind) => {
-        return this.createTemplateInstance(secret).then(() => {
-          this.setState({ inProgress: false });
-          const activeExtension = perspectiveExtensions.find(
-            (p) => p.properties.id === activePerspective,
-          );
-          (async () => {
-            const url = (await activeExtension.properties.importRedirectURL())(namespace);
-            history.push(url);
-          })();
-        });
+        return this.createTemplateInstance(secret).then(
+          async (templateInstance: K8sResourceKind) => {
+            await this.updatesecretOwnerRef(secret, templateInstance);
+            this.setState({ inProgress: false });
+            const activeExtension = perspectiveExtensions.find(
+              (p) => p.properties.id === activePerspective,
+            );
+            (async () => {
+              const url = (await activeExtension.properties.importRedirectURL())(namespace);
+              history.push(url);
+            })();
```

Comment:
I just noticed this, we won't need the async iife anymore since the entire function is async, would be nice to remove it.